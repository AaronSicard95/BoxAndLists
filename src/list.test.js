import { fireEvent, queryByText, render } from "@testing-library/react";
import ListForm from "./listForm";
import TodoList from "./todoList";

it("check render", function(){
    render(<ListForm/>);
});

it("match snap", function(){
    const {asFragment, getByLabelText, queryByText} = render(<ListForm/>);
    expect(asFragment()).toMatchSnapshot();
    expect(queryByText("Go Shopping")).not.toBeInTheDocument();
    //course says to use name, but had to change to ID
    //to prevent error being thrown. Why?
    fireEvent.change(getByLabelText("ToDo:"), {target:{value:"Go Shopping"}})
    fireEvent.click(queryByText("Add ToDo!"));
    expect(queryByText("Go Shopping")).toBeInTheDocument();
})