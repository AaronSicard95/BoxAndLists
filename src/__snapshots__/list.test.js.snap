// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snap 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
        ToDo: 
      </label>
      <input
        id="todo"
        type="text"
        value=""
      />
      <button>
        Add ToDo!
      </button>
    </form>
    <div
      style="display: inline;"
    />
  </div>
</DocumentFragment>
`;
